ConfigKey.go

```diff
@@ -67,16 +68,19 @@ func (configKey *ConfigKey) CheckRequired() error {
 // Return an error on validation failure of one of the validator.
 //
 // Return on first failure, the remaining validator are not ran.
-func (configKey *ConfigKey) CheckValidators() error {
+func (configKey *ConfigKey) CheckValidators() []error {
```

Comment:
Why don't you add the []errrors in the parameters instead of  return it ?

Reply:
It could be done that way, but here it's explicit that we get the errors as a return value. Passing an error slice would expose the function to some potentials problem such as getting a nil slice. This situation could be dealt with but we can avoid it altogether by returning the error slice as done here.